pyscf/lib/vhf/nr_sgx_direct.c

```diff
@@ -35,138 +36,223 @@ typedef struct {
+                                    i0, i1, j0, j1, inds, tmp_ngrids);
+        }
+        
+        free(inds);
 }
 
 
 void SGXnr_direct_drv(int (*intor)(), void (*fdot)(), SGXJKOperator **jkop,
-                      double **dms, double **vjk, int n_dm, int ncomp,
-                      int *shls_slice, int *ao_loc,
-                      CINTOpt *cintopt, CVHFOpt *vhfopt,
-                      int *atm, int natm, int *bas, int nbas, double *env)
+                        double **dms, double **vjk, int n_dm, int ncomp,
+                        int *shls_slice, int *ao_loc,
+                        CINTOpt *cintopt, CVHFOpt *vhfopt,
+                        int *atm, int natm, int *bas, int nbas, double *env,
+                        int env_size, int aosym)
 {
-        IntorEnvs envs = {natm, nbas, atm, bas, env, shls_slice, ao_loc, NULL,
-                cintopt, ncomp};
 
-        const int ksh0 = shls_slice[4];
-        const int ksh1 = shls_slice[5];
-        int nksh = ksh1 - ksh0;
+        const int ish0 = shls_slice[0];
+        const int ish1 = shls_slice[1];
+        const int jsh0 = shls_slice[2];
+        const int jsh1 = shls_slice[3];
+        int nish = ish1 - ish0;
         int di = GTOmax_shell_dim(ao_loc, shls_slice, 2);
-        int cache_size = GTOmax_cache_size(intor, shls_slice, 2,
-                                           atm, natm, bas, nbas, env);
+        int cache_size = _max_cache_size_sgx(intor, shls_slice, 2,
+                                             atm, natm, bas, nbas, env);
+        int npair;
+        if (aosym == 2) {
+            npair = nish * (nish+1) / 2;
+        } else {
+            npair = nish * nish;
+        }
+
+        int (*fprescreen)();
+        if (vhfopt) {
+                fprescreen = vhfopt->fprescreen;
+        } else {
+                fprescreen = CVHFnoscreen;
+        }
+        int ngrids = env[NGRIDS];
+        double* all_grids = env+(size_t)env[PTR_GRIDS];
 
 #pragma omp parallel default(none) \
         shared(intor, fdot, jkop, ao_loc, shls_slice, \
-               dms, vjk, n_dm, ncomp, nbas, vhfopt, envs, \
-               nksh, di, cache_size)
+               dms, vjk, n_dm, ncomp, nbas, vhfopt, \
+               atm, bas, env, natm, \
+               nish, di, cache_size, fprescreen, \
+               aosym, npair, cintopt, env_size, \
+               ngrids, all_grids)
 {
-        int i, ksh;
+        int i, ij, ish, jsh;
+        int shls[4];
+        double* tmp_env = (double*) malloc(env_size * sizeof(double));
+        for (i = 0; i < env_size; i++) {
+            tmp_env[i] = env[i];
+        }
+        IntorEnvs envs = {natm, nbas, atm, bas, tmp_env, shls_slice, ao_loc, NULL,
+                          cintopt, ncomp};
         SGXJKArray *v_priv[n_dm];
         for (i = 0; i < n_dm; i++) {
-                v_priv[i] = jkop[i]->allocate(shls_slice, ao_loc, ncomp);
+                v_priv[i] = jkop[i]->allocate(shls_slice, ao_loc, ncomp, env[NGRIDS]);
```

Comment:
do not use env[NGRIDS] here, it's wrong type

Reply:
Fixed by 4ef2ea7